kinto_fxa/scripts/process_account_events.py

```diff
@@ -0,0 +1,122 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+"""Script to process account-related events from an SQS queue.
+
+This script polls an SQS queue for events indicating activity on an upstream
+account, as documented here:
+
+  https://github.com/mozilla/fxa-auth-server/blob/master/docs/service_notifications.md
+
+The following event types are currently supported:
+
+  * "delete": the account was deleted; we delete their default bucket
+    to comply with GDPR. Note that this won't be sufficient in
+    applications where users can store data in locations besides the
+    default bucket.
+
+Note that this script may not be necessary in all applications of
+kinto-fxa.
+
+"""
+
+import itertools
+import json
+import logging
+import uuid
+
+import boto3
+from ec2_metadata import ec2_metadata
+from kinto.core.utils import hmac_digest
+import transaction as current_transaction
+
+logger = logging.getLogger(__name__)
+
+
+def get_default_bucket_id(config, uid):
+    secret = config['registry'].settings['userid_hmac_secret']
+    digest = hmac_digest(secret, uid)
+    return str(uuid.UUID(digest[:32]))
```

Comment:
I decided not to use the `default_bucket` plugin's implementation of this function because it takes a request, which we don't really have. I decided not to refactor the function to expose a "pure" version of this calculation in the interests of time (don't want to cut a new release of kinto, etc.). It's a small enough function that I didn't think it was a big deal to copy it, but I'm willing to hear other opinions.